rclcpp/src/rclcpp/signal_handler.cpp

```diff
@@ -113,7 +113,7 @@ SignalHandler::get_logger()
 SignalHandler &
 SignalHandler::get_global_signal_handler()
 {
-  static SignalHandler signal_handler;
+  static SignalHandler & signal_handler = *new SignalHandler();
```

Comment:
I believe that memory debuggers are aware of this pattern. They actually won't report this as a memory leak because there's always a valid reference to it during execution until final termination due to the statically allocated reference. As such for all times during the program's execution this memory is reachable.

Reply:
There's no good time to delete it, and though I don't love it either, it is the recommended solution in the google c++ style guide. 🤷

Reply:
The alternative would be to ensure this class is trivially destructible which isn't possible given that it needs to store things like the `rclcpp::Logger` which itself stores a string, a `std::thread`, etc... So I think this is the best solution unfortunately.

I will point out it is a recommended style, not a correctness thing. What's there today is valid and (as far as we can see) correct code, but the issue is if a bug works its way in, this pattern (non-trivial static objects) makes diagnosing harder, hence the recommendation. So I think it's a valid thing to change.